torch/_export/graph_module.py

```diff
@@ -224,61 +193,96 @@ def op_to_str(op):
         return (reduce_graph_module, (pickled_state,))
 
 
-def attach_export_graph_metadata(gm: fx.GraphModule, meta: ExportMetadata) -> None:
-    gm.meta[EXPORT_METADATA] = meta
-    gm.__class__ = type(type(gm).__name__, (ExportGraphModuleMixin, type(gm)), {})
-
-
 def make_export_graph_module(
     root: Union[torch.nn.Module, Dict[str, Any]],
-    graph: fx.Graph,
-    in_spec: Optional[pytree.TreeSpec] = None,
-    out_spec: Optional[pytree.TreeSpec] = None,
-    mutation: Optional[List[Tuple[str, List[str]]]] = None,
-    example_inputs: Any = None,
+    graph: torch.fx.Graph,
     class_name: str = "ExportGraphModule",
-) -> fx.GraphModule:
-
-    input_shape_constraints = []
-    inline_constraints = {}
-    if isinstance(root, torch.fx.GraphModule):
-        input_shape_constraints = root.meta.get("input_shape_constraints", [])
-        inline_constraints = root.meta.get("inline_constraints", {})
-
-    gm = fx.GraphModule(root, graph, class_name)
-
-    input_tracker = 0
-
-    # group by input id
-    input_shape_constraints_by_tensor_id = defaultdict(list)
-    for constraint in input_shape_constraints:
-        input_shape_constraints_by_tensor_id[constraint["t_id"]].append((constraint["dim"], constraint["min"], constraint["max"]))
-
-    input_shape_constraints_by_src_name: Dict[str, List[Tuple[int, ConstraintExpr, ConstraintExpr]]] = {}
-    input_name_to_example_inputs: Dict[str, Any] = {}
-    if example_inputs is not None:
-        input_tracker = 0
-        for node in gm.graph.nodes:
-            if node.op == "placeholder":
-                example_input = example_inputs[input_tracker]
-                if id(example_input) in input_shape_constraints_by_tensor_id:
-                    input_shape_constraints_by_src_name[node.name] = input_shape_constraints_by_tensor_id[id(example_input)]
-                input_name_to_example_inputs[node.name] = example_input
-                input_tracker += 1
-
-    meta = ExportMetadata(
-        in_spec=in_spec,
-        out_spec=out_spec,
-        update_spec=0,
-        mutation=mutation if mutation else [],
-        input_shape_constraints=input_shape_constraints_by_src_name,
-        inline_constraints=inline_constraints,
-        input_name_to_example_inputs=input_name_to_example_inputs,
-    )
-    attach_export_graph_metadata(gm, meta)
+) -> torch.fx.GraphModule:
+    gm = torch.fx.GraphModule(root, graph, class_name=class_name)
+    gm.__class__ = type(type(gm).__name__, (ExportGraphModuleMixin, type(gm)), {})
     return gm
 
 
+class ExportedProgram:
+    def __init__(
+        self,
+        root: Union[torch.nn.Module, Dict[str, Any]],
+        graph: torch.fx.Graph,
+        graph_signature: Optional[ExportGraphSignature] = None,
+        call_spec: Optional[CallSpec] = None,
+        state_dict: Optional[Dict[str, Any]] = None,
```

Comment:
Why these are optional?

Reply:
Hmm, I just set it as optional until Tugsuu populates it with the results of aot_export. I guess I can just set it as non-optional and pass in empty values?